src/features/payment-agreements/hooks/mutations/use-create-payment-agreement-mutation.ts

```diff
@@ -19,6 +19,7 @@ export function useCreatePaymentAgreementMutation() {
     await client.invalidateQueries({queryKey: accountBalancesQueryKey(address)})
     await client.invalidateQueries({queryKey: accountTokenOperationsQueryKey(address)})
     await client.invalidateQueries({queryKey: accountPaymentAgreementsCreatedQueryKey(address)})
+    await client.resetQueries({queryKey: accountPaymentAgreementsCreatedQueryKey(address), exact: true})
```

Comment:
Why do we need `await client.resetQueries(...)` if we already have `await client.invalidateQueries(...)` for the same key?

Reply:
Becasuse await client.invalidateQueries(...) doesn't do what I needed it to do. await client.resetQueries(...) does.

Reply:
Both `await client.invalidate` and `await client.resetQueries` both invalidates the data. The reason why it does not refetch the data is because we have `refetchOnMount: true`.

 `client.invalidate` should be preferred because it offers a better user experience.